tests/frameworks/test_fastapi.py

```diff
@@ -1,628 +1,586 @@
 # (c) Copyright IBM Corp. 2021
 # (c) Copyright Instana Inc. 2020
 
-import time
-import unittest
-import multiprocessing
-
-import requests
-
-from instana.singletons import async_tracer
-from tests.apps.fastapi_app import launch_fastapi
-from ..helpers import testenv
-from ..helpers import get_first_span_by_filter
-
-
-class TestFastAPI(unittest.TestCase):
-    def setUp(self):
-        self.proc = multiprocessing.Process(target=launch_fastapi, args=(), daemon=True)
-        self.proc.start()
-        time.sleep(2)
-
-    def tearDown(self):
-        # Kill server after tests
-        self.proc.kill()
-
-    def test_vanilla_get(self):
-        result = requests.get(testenv["fastapi_server"] + "/")
-
-        self.assertEqual(result.status_code, 200)
-        self.assertIn("X-INSTANA-T", result.headers)
-        self.assertIn("X-INSTANA-S", result.headers)
-        self.assertIn("X-INSTANA-L", result.headers)
-        self.assertEqual(result.headers["X-INSTANA-L"], "1")
-        self.assertIn("Server-Timing", result.headers)
-
-        spans = async_tracer.recorder.queued_spans()
-        # FastAPI instrumentation (like all instrumentation) _always_ traces unless told otherwise
-        self.assertEqual(len(spans), 1)
-        self.assertEqual(spans[0].n, "asgi")
-
-    def test_basic_get(self):
+from typing import Generator
+
+from fastapi.testclient import TestClient
+import pytest
+from instana.singletons import tracer, agent
+
+from tests.apps.fastapi_app.app import fastapi_server
+from tests.helpers import get_first_span_by_filter
+
+
+class TestFastAPI:
+    @pytest.fixture(autouse=True)
+    def _resource(self) -> Generator[None, None, None]:
+        """SetUp and TearDown"""
+        # setup
+        # We are using the TestClient from Starlette/FastAPI to make it easier.
+        self.client = TestClient(fastapi_server)
+
+        # Clear all spans before a test run
+        self.recorder = tracer.span_processor
+        self.recorder.clear_spans()
+
+        # Hack together a manual custom headers list; We'll use this in tests
+        agent.options.extra_http_headers = [
+            "X-Capture-This",
+            "X-Capture-That",
+            "X-Capture-This-Too",
+            "X-Capture-That-Too",
+        ]
+        yield
```

Comment:
Since we don't have a teardown logic here and `autouse` is set to `True`, I think we can safely remove the keyword `yield` here.
```suggestion
```

Reply:
Done